citellusclient/plugins/core/network/interface-onboot.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+# Copyright (C) 2020 Volkan Yalcin <vlyalcin@gmail.com>
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# long_name: Reports interfaces that its state 'up' and doesn't have ONBOOT='yes' in its network-scripts file.
+# description: Reports interfaces that its state 'up' and doesn't have ONBOOT='yes' in its network-scripts file.
+# bugzilla: https://github.com/citellusorg/citellus/issues/535
+# priority: 100
+# kb:
+
+# Test cases
+# 1 Give an invalid SOS_IP_ADDRESS_PATHS. Observe the fail.
+# 2 Remove ONBOOT=yes from network file for the UP interfaces. Observe the fail.
+
+# Load common functions
+[[ -f "${CITELLUS_BASE}/common-functions.sh" ]] && . "${CITELLUS_BASE}/common-functions.sh"
+
+NETWORK_SCRIPTS_PATH="/etc/sysconfig/network-scripts/ifcfg"
+
+if [[ ${CITELLUS_LIVE} -eq 0 ]]; then
+    NETWORK_SCRIPTS_PATH="${CITELLUS_ROOT}${NETWORK_SCRIPTS_PATH}"
+    IP_ADDRESS_FILE=$(first_file_available "${CITELLUS_ROOT}/sos_commands/networking/ip_-d_address" "${CITELLUS_ROOT}/sos_commands/networking/ip_address")
+    is_required_file "${IP_ADDRESS_FILE}"
+elif [[ ${CITELLUS_LIVE} -eq 1 ]]; then
+    IP_ADDRESS_FILE=$(mktemp)
+    trap "rm ${IP_ADDRESS_FILE}" EXIT
+    ip address  > ${IP_ADDRESS_FILE} 2>&1
+fi
+
+RC_STATUS=${RC_OKAY}
+for interface_name in $(grep -i "state UP" ${IP_ADDRESS_FILE} |cut -f2 -d ":"); do
+    NETWORK_INTERFACE_FILE="${NETWORK_SCRIPTS_PATH}-${interface_name}"
+    if ! grep -iq "onboot=yes" "${NETWORK_INTERFACE_FILE}"; then
```

Comment:
If the  prior file doesn't exist, execution gives an uncaught error:

 grep: /etc/sysconfig/network-scripts/ifcfg-wlp2s0: No such file or directory


Probably would be good to check if file exists and fail if either the file is missing or if it doesn't contain onboot=yes

Reply:
I thought to use _is_required_file_ macro but it interrupts the execution and skips the rest of the code. So it blocks checking all interfaces.

>  /home/volkan/PycharmProjects/citellus/citellusclient/plugins/core/network/interface-onboot.sh: skipped
>     required file /etc/sysconfig/network-scripts/ifcfg-enp4s0f0 not found.

So I think I need to write separate "if" control just above the grep command. I said separate because that will help us to raise correct fail message. Otherwise, we will print the same message for cases that doesn't contain "onboot=yes" and file does not exist.

What do you think?

Reply:
Yes, separate if would be the best or capture the error on grep and redirect to /dev/null

Reply:
I have added separate "if". Then I have added -s parameter to grep according to information at link https://stackoverflow.com/a/6426401/1176555

By the way, please feel free to change any file name, error message or parameter name if you feel it can be better.

Thanks.